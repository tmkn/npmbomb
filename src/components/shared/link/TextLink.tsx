/** @jsx jsx */
import { jsx, css } from "@emotion/react";
import React from "react";

import { mq, secondaryColor, primaryColor } from "../../../css";

export const TextLink: React.FC<{ href: string; target?: string }> = ({
    children,
    href,
    target
}) => {
    const style = css({
        [mq[0]]: {
            textDecoration: "underline",
            textDecorationColor: secondaryColor,
            color: primaryColor,
            cursor: "pointer"
        }
    });

    return (
        <a css={style} href={href} target={target ?? "_blank"}>
            {children}
        </a>
    );
};

export const ClickLink: React.FC<{ onClick: () => void }> = ({ children, onClick }) => {
    const style = css({
        [mq[0]]: {
            textDecoration: "underline",
            textDecorationColor: secondaryColor,
            color: secondaryColor,
            cursor: "pointer"
        }
    });

    return (
        <span css={style} onClick={onClick}>
            {children}
        </span>
    );
};
