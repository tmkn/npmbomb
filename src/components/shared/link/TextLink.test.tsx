import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";

import { TextLink } from "./TextLink";

describe("<TextLink />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(<TextLink href="abc">Hello World</TextLink>);

        expect(asFragment()).toMatchSnapshot();
    });

    test("render basic link", async () => {
        const { findAllByText, getByText } = render(<TextLink href="abc">Hello World</TextLink>);

        expect(getByText("Hello World")).toBeDefined();
    });
});
