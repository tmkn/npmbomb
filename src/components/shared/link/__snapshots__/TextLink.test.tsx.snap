// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLink /> matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="css-1qgk5q"
    href="abc"
  >
    Hello World
  </a>
</DocumentFragment>
`;
